src/presenced.cpp

```diff
@@ -350,6 +350,8 @@ Client::reconnect(const std::string& url)
                     mConnectPromise.reject(err.msg(), err.code(), WS_ERRTYPE_DNS);
                     mLoginPromise.reject(err.msg(), err.code(), WS_ERRTYPE_DNS);
                 }
+
+                return err;
```

Comment:
I think this case is quite different from previous. This promise-fail is handled at chatClient.cpp:718